kong/dao/schemas/apis.lua

```diff
@@ -168,11 +168,20 @@ local function check_name(name)
 end
 
 -- check that retries is a valid number
-local function check_retries(retries)
+local function check_smallint(i)
   -- Postgres 'smallint' size, 2 bytes
-  if (retries < 0) or (math.floor(retries) ~= retries) or (retries > 32767) then
-    return false, "retries must be an integer, from 0 to 32767"
+  if i < 0 or math.floor(i) ~= i or i > 32767 then
+    return false, "must be an integer between 0 and 32767"
   end
+
+  return true
+end
+
+local function check_u_int(t)
+  if t < 1 or t > 2^31 - 1 or math.floor(t) ~= t then
+    return false, "must be an integer between 1 and 2^31 - 1"
```

Comment:
error message to technical. And in general (matter of taste) I prefer this; https://github.com/Mashape/kong/blob/a6f0cb38661a09cd91802fbdd4ba8c1c696b17d0/kong/dao/schemas/upstreams.lua#L4-L6